### PR TITLE
BUGFIX: Fix subtree tagging after move

### LINKED ISSUE #5618: BUG: Moved nodes are tagged with subtree based on previous parent node and thus lost
Steps to reproduce:

From a multiple column element move all text elements out of it directly into the content collection (neos demo / feature / "What Makes Neos Special")

This leaves us with an empty column element:

Image

Now remove this column element. At first the neos ui only optimistically removes that, but after a reload the content repository also shows that the extracted text nodes are missing.

Note that i could not find a work around to circumvent this. Publishing before removing the columns does not work and just delays this fate. Also moving the nodes on another part onto the document or into another document does not stop this behavior.

The reason for this can be seen in the projection. The 6 events `NodeAggregateWasMoved` and the single  `SubtreeWasTagged` are correct.
The first Headline node has a relation anchor of `82` and the columns element has `76`.

The hierarchy relations for the editors workspace and language show that the subtreetag is not only placed on the columns but also inherited by the other contents which now live directly in the main content collection (`35`)

| position | parentnodeanchor | childnodeanchor | subtreetags |
| :--- | :--- | :--- | :--- |
| 0 | 35 | 82 | {"removed": null} |
| 64 | 35 | 83 | {"removed": null} |
| 96 | 35 | 84 | {"removed": null} |
| 112 | 35 | 85 | {"removed": null} |
| 120 | 35 | 86 | {"removed": null} |
| 124 | 35 | 87 | {"removed": null} |
| 128 | 35 | 76 | {"removed": true} |
| 256 | 35 | 77 | {} |
| 384 | 35 | 78 | {} |
| 128 | 35 | 76 | {"removed": true} |
| 256 | 35 | 77 | {} |
| 384 | 35 | 78 | {} |

### PR DESCRIPTION
**Pull request by nezaniel**
Resolves: #5618

Previously, when cascading subtree tags (both add and remove), the whole set of affected DSPs was evaluated on each level.

This affected also hierarchy relations not connected to the tagged node in their DSP (see #5618)

Visualization:
TagAfterMove

**Upgrade instructions**

Neos uses subtree tags (for disabling and soft removal) always in "allSpecializations" mode, i.e. generalizations and peer variants are unaffected.

Thus, if (and only if) a project has specialization variants, it is affected by the underlying bug anyway. In that case, a replay of the content graph projection is advised.

**Review instructions**

I've done an edge case analysis for the new test cases, please check them for completeness and that the step definitions match the respective parameters.

**Checklist**

- [X] Code follows the PSR-12 coding style
- [X] Tests have been created, run and adjusted as needed
- [X] The PR is created against the 9.0 branch
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/08-NodeMove/08-SubtreeTaggingAfterNodeMove.feature [hunk 1]
```diff
@@ -0,0 +1,1327 @@
...
+  - scatter
+  - gatherSpecializations
+  - gatherAll
+  strategy and then
+  - tag
+  - untag
+  the parent in
+  - allSpecializations
+  - allVariants
+
```
Comment (skurfuerst) [inline Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/08-NodeMove/08-SubtreeTaggingAfterNodeMove.feature:34]:
```suggestion
  Idea for testing TAGGING ("... then tag the parent ..."):
  - lady-eleonode-rootford
  -- sir-david-nodenborough  <-- (2) Tag
  --- nody-mc-nodeface       <-- (1) Move Source
  ---- nodimus-mediocre
  -- sir-nodeward-nodington-iii
  --- nodimus-prime
  --- .....................  <-- (1) Move Target

  Idea for testing UNTAGGING ("... then untag the parent ..."):
  - lady-eleonode-rootford
  -- sir-david-nodenborough  <-- (1) Tag         (3) untag
  --- nody-mc-nodeface       <-- (1) Move Source
  ---- nodimus-mediocre
  -- sir-nodeward-nodington-iii <-- (1) Tag
  --- nodimus-prime
  --- .....................  <-- (1) Move Target

```

**Review by skurfuerst — APPROVED** [review]
I understand the problem and the high-level solution idea from the PR.

I don't fully grasp the implementation, but I am very confident in the testcase (which I looked quite deeply into for about ~1h now).

GREAT WORK <3 <3 <3

**Review by mhsdesign — COMMENTED** [review]
Post merge +1 thanks especially for the visual graph and for all the test combinations!

The solution makes perfect sense concept wise (if i would understand this sql then probably that too ^^)
